server/src/main/java/com/cloud/resourcelimit/ResourceLimitManagerImpl.java

```diff
@@ -1171,6 +1178,27 @@ public ResourceCountCheckTask() {
 
         @Override
         protected void runInContext() {
+            GlobalLock lock = GlobalLock.getInternLock("ResourceCheckTask");
+            try {
+                if (lock.lock(30)) {
+                    ManagementServerHostVO msHost = managementServerHostDao.findOneByLongestRuntime();
+                    if (msHost == null || (msHost.getMsid() != ManagementServerNode.getManagementServerId())) {
+                        s_logger.debug("Skipping the resource counters recalculation task on this management server");
```

Comment:
maybe make this a trace message?
```suggestion
                        s_logger.trace("Skipping the resource counters recalculation task on this management server");
```